logging/configure.go

```diff
@@ -0,0 +1,105 @@
+package logging
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/armory-io/monitoring/log/formatters"
+	"github.com/armory-io/monitoring/log/hooks"
+	"github.com/sirupsen/logrus"
+)
+
+type Config struct {
+	Remote RemoteLoggingConfig `json:"remote" yaml:"remote"`
+	JSON   FormatJson          `json:"json" yaml:"json"`
+}
+
+type RemoteLoggingConfig struct {
+	Enabled    bool   `json:"enabled" yaml:"enabled"`
+	Endpoint   string `json:"endpoint" yaml:"endpoint"`
+	Version    string `json:"version" yaml:"version"`
+	CustomerID string `json:"customerId" yaml:"customerId"`
+}
+
+type FormatJson struct {
+	Enabled bool              `json:"enabled" yaml:"enabled"`
+	Fields  map[string]string `json:"fields" yaml:"fields"`
+}
+
+func (fj *FormatJson) Configure(l *logrus.Logger) {
+	// TODO - support configuration of field names
+	fm := logrus.FieldMap{}
+	for f, v := range fj.Fields {
+		// we have to switch here because setting fm[f] is of type
+		// fieldKey, not a string...
+		switch f {
+		case logrus.FieldKeyTime:
+			fm[logrus.FieldKeyTime] = v
+		case logrus.FieldKeyMsg:
+			fm[logrus.FieldKeyMsg] = v
+		case logrus.FieldKeyLevel:
+			fm[logrus.FieldKeyLevel] = v
+		}
+	}
+	formatter := logrus.JSONFormatter{FieldMap: fm}
+	l.SetFormatter(&formatter)
+
+}
+
+func ConfigureLogrus(l *logrus.Logger, config Config) error {
```

Comment:
Could we rename so the implementation is not attached to a specific library if we want to swap later on? Would it be worth providing our own interface for `logrus.Logger` that new work would be using?

Having a `NoopLogger` would be nice for testing as well. I can't remember if logrus had any out of the box.

Reply:
@ncknt - Are you suggesting that this library defines the underlying log library instead of the consumer? I.e. I let this library dictate which one I'm using instead of choosing it for myself?

Reply:
@dogonthehorizon - any thoughts on that, assuming I'm interpreting correctly.

Reply:
Yes. The reasoning is to have something similar to slf4j and use a facade. It could be as simple as a `Logger` interface. If we find a better library than logrus, it would be trivial to swap implementation.